src/us/kbase/assemblyhomology/storage/mongo/MongoAssemblyHomologyStorage.java

```diff
@@ -396,6 +399,7 @@ public Namespace getNamespace(final NamespaceID namespaceID)
 
 	private Namespace toNamespace(final Document ns) throws AssemblyHomologyStorageException {
 		try {
+			final String fid = ns.getString(Fields.NAMESPACE_FILTER_ID);
```

Comment:
I only do javadocs for functions that will show up in built docs, so `public` or `protected`.

For other functions, the user must be looking at the code and my philosophy is that the code should be self-documenting - ideally the function signature tells you what the function does. In this case it's pretty obvious - it takes a `Document` returned from mongo and turns it into a `Namespace`.

Of course, sometimes comments are needed for bits of code that may be more complicated or less intuitive. Obviously I think I'm commenting where it's needed, but if you see something specific where it's not clear what's going on, let me know.

Reply:
> Ok, so if how about just plain old comments then?

That's really no different than javadocs. You just omit a few @params, etc. 

Ideal code is self documenting - the function signature explains what the method does and the code should be readable enough that any documentation would be redundant. Documentation can be harmful - it can be incorrect and it doesn't update automatically when the code is updated. APIs must be documented, obviously, but implementation details do not unless the code is tricky or unintuitive, because the code itself is the documentation. Documentation becomes a crutch or excuse for sloppy, illegible code in the worst case.

> I was specifically instructed during standup to make sure that the code I review (even private code) gets commented, so that's why I am bringing it up. ("Documentation: All methods are commented in clear language.")

I can't imagine the rule is intended to mean methods that are hidden implementation details. If it is we need to change that. The rule should be that hidden code is clear enough that documentation is not required, and iff the code cannot be made clear enough (which should be relatively rare) then comments are added at the point of lack of clarity.